File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/payment/UserCardListAdapter.java
package com.lattis.ellipse.presentation.ui.payment;


import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.lattis.ellipse.domain.model.Card;
import com.lattis.ellipse.presentation.ui.bike.BikeInfoActivity;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.lattis.ellipse.R;

public class UserCardListAdapter extends RecyclerView.Adapter<UserCardListAdapter.ViewHolder> {
    List<Card> cardList;
    Context mContext;
    boolean isSetClickEvent;
    UserCardListListener userCardListListener;


    public UserCardListAdapter(Context context, List<Card> cards, boolean isShowCheckbox, UserCardListListener listener) {
        this.mContext = context;
        this.cardList = cards;
        this.isSetClickEvent = isShowCheckbox;
        this.userCardListListener = listener;

    }


    @Override
    public UserCardListAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.user_card_view, parent, false);
        return new UserCardListAdapter.ViewHolder(view, cardList, isSetClickEvent);
    }

    @Override
    public void onBindViewHolder(UserCardListAdapter.ViewHolder holder, int position) {
        if (cardList != null) {
            holder.tv_cardType.setText(cardList.get(position).getCc_type());
            String number = cardList.get(position).getCc_no();
            number = number.substring(number.length() - 8);
            number = number.substring(0, number.length() - 4) + "\n" + number.substring(number.length() - 4, number.length()); // 18:00
            holder.tv_cardNumber.setText(number);
            String cardName = cardList.get(position).getCc_type().toUpperCase().replace(" ", "_");
            holder.card_number_icon.setImageResource(getResource(cardName));
            holder.checkbox.setChecked(cardList.get(position).getIs_primary());
            holder.checkbox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    if (isChecked) {
                        userCardListListener.onclickCheckBox(position);
                    } else {
                        holder.checkbox.setChecked(true);
                    }

                }
            });

        }


    }

    @Override
    public int getItemCount() {
        return cardList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        @BindView(R.id.tv_cardType)
        TextView tv_cardType;
        @BindView(R.id.tv_cardNumber)
        TextView tv_cardNumber;
        @BindView(R.id.checkbox)
        CheckBox checkbox;
        @BindView(R.id.card_number_icon)
        ImageView card_number_icon;
        List<Card> list;

        public ViewHolder(View itemView, List<Card> cardList, boolean isSetClickEvent) {
            super(itemView);
            if (isSetClickEvent)
                itemView.setOnClickListener(this);
            this.list = cardList;
            ButterKnife.bind(this, itemView);
        }


        @Override
        public void onClick(View v) {
            v.getContext().startActivity(new Intent(v.getContext()
                    , AddCardActivity.class).putExtra("CARD_DETAILS",
                    new Gson().toJson(list.get(getAdapterPosition()))));
        }
    }

    private int getResource(String card_type) {
        switch (card_type) {
            case "VISA":
                return R.drawable.bt_ic_visa;

            case "MASTERCARD":
                return R.drawable.bt_ic_mastercard;

            case "DISCOVER":
                return R.drawable.bt_ic_discover;

            case "AMERICAN_EXPRESS":
                return R.drawable.bt_ic_amex;

            case "MAESTRO":
                return R.drawable.bt_ic_maestro;

            case "DINERS_CLUB":
                return R.drawable.bt_ic_diners_club;

            case "UNIONPAY":
                return R.drawable.bt_ic_unionpay;

            case "JCB":
                return R.drawable.bt_ic_jcb;
        }

        return R.drawable.bt_ic_unknown;
    }

}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/payment/UserCardListAdapter.kt
package com.lattis.lattis.presentation.payment

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.lattis.domain.models.Card
import com.lattis.lattis.presentation.customview.CustomTextView
import io.lattis.lattis.R

class UserCardListAdapter(
    var mContext: Context,
    var cardList: List<Card>?,
    var isSetClickEvent: Boolean,
    var userCardListListener: PaymentCardClickListener
) : RecyclerView.Adapter<UserCardListAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.activity_payment_item_card_list, parent, false)
        return ViewHolder(
            view,
            isSetClickEvent
        )
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        if (cardList != null) {
            var number = cardList!![position].cc_no
            holder.ct_card_number_in_payment_card_list!!.text = cardList!!.get(position).cc_type+" "+number!!.substring(number.length - 4, number.length)
            val cardName = cardList!![position].cc_type!!.toUpperCase().replace(" ", "_")
            holder.iv_card_list_item!!.setImageResource(getResource(cardName))
            holder.iv_check_mark_in_payment_card_list!!.isChecked = cardList!![position].getIs_primary()
            holder.iv_check_mark_in_payment_card_list!!.setOnCheckedChangeListener { buttonView, isChecked ->
                if (isChecked) {
                    userCardListListener.onCardCheckBoxClicked(position)
                } else {
                    holder.iv_check_mark_in_payment_card_list!!.isChecked = true
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return cardList!!.size
    }

    inner class ViewHolder(
        itemView: View,
        isSetClickEvent: Boolean
    ) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {

        var iv_card_list_item:ImageView = itemView.findViewById(R.id.iv_card_list_item)
        var ct_card_number_in_payment_card_list:CustomTextView = itemView.findViewById(R.id.ct_card_number_in_payment_card_list)
        var iv_check_mark_in_payment_card_list : CheckBox = itemView.findViewById(R.id.iv_check_mark_in_payment_card_list)


        override fun onClick(v: View) {
            userCardListListener.onCardClicked(adapterPosition)
        }

        init {
            if (isSetClickEvent) itemView.setOnClickListener(this)
        }
    }

    private fun getResource(card_type: String): Int {
        when (card_type) {
            "VISA" -> return R.drawable.bt_ic_visa
            "MASTERCARD" -> return R.drawable.bt_ic_mastercard
            "DISCOVER" -> return R.drawable.bt_ic_discover
            "AMERICAN_EXPRESS" -> return R.drawable.bt_ic_amex
            "MAESTRO" -> return R.drawable.bt_ic_maestro
            "DINERS_CLUB" -> return R.drawable.bt_ic_diners_club
            "UNIONPAY" -> return R.drawable.bt_ic_unionpay
            "JCB" -> return R.drawable.bt_ic_jcb
        }
        return R.drawable.bt_ic_unknown
    }

}